ygot/struct_validation_map.go

```diff
@@ -600,15 +614,22 @@ func copyPtrField(dstField, srcField reflect.Value) error {
 // copyInterfaceField copies srcField into dstField. Both srcField and dstField
 // are reflect.Value structs which contain an interface value.
 func copyInterfaceField(dstField, srcField reflect.Value) error {
-	if srcField.IsNil() || !srcField.IsValid() {
+	if util.IsNilOrInvalidValue(srcField) {
 		return nil
 	}
 
 	if !util.IsValueInterface(srcField) || !util.IsValueStructPtr(srcField.Elem()) {
 		return fmt.Errorf("invalid interface type received: %T", srcField.Interface())
 	}
 
-	s := srcField.Elem().Elem() // Dereference to a struct.
+	s := srcField.Elem().Elem() // Dereference src to a struct.
+	if !util.IsNilOrInvalidValue(dstField) {
+		dV := dstField.Elem().Elem() // Dereference dst to a struct.
+		if !reflect.DeepEqual(s.Interface(), dV.Interface()) {
```

Comment:
What's the reason to not use `cmp` here?

Reply:
AFAIK, the only "recommended" move from `reflect.DeepEqual` to `cmp.Equal` is for proto messages, since `reflect.DeepEqual` doesn't work well for protos. In general, I've tried to use `reflect.DeepEqual` where we don't need the flexibility of `cmp`.